tests/test_api.py

```
@@ -32,6 +65,42 @@ def test_query_post_returns_json():
+    assert data["schema_version"] == "1"
+    assert data["query"] == "test"
+    assert data["mode"] == "hybrid"
+    assert isinstance(data["results"], list)
+    assert data["capabilities"]["json_query"] is True
+
+
+def test_api_search_normalizes_raw_query_shape(monkeypatch):
+    monkeypatch.setattr(
+        api_module,
+        "_dispatch_query",
+        lambda q, k, mode="semantic", collections=None: {
```

Ruff (0.15.13)

[warning] Unused lambda argument: `q`

(ARG005)